community/kernel/src/main/java/org/neo4j/kernel/impl/api/dbms/NonTransactionalDbmsOperations.java

```diff
@@ -47,9 +47,11 @@ public NonTransactionalDbmsOperations( Procedures procedures )
     ) throws ProcedureException
     {
         BasicContext ctx = new BasicContext();
-        if ( mode instanceof AuthSubject )
+
+        AccessMode originalMode = mode.getOriginalAccessMode();
```

Comment:
We are only using it to get hold of the `AuthSubject` through a downcast. The only permission checks done in dbms procedures are done directly on auth subject, like `isAdmin()` or has a username.
We have a refactoring planned of separating access mode and auth subject into a security context, but right now you get auth subject through the reference of the access mode.